File: server/models/tournament.py
from pydantic import BaseModel, Field
from typing import List
from services.database import default_id

class Tournament(BaseModel):
    id: str = Field(default_factory=default_id, alias="_id")
    owner: str
    participants: List[str]
    confirmed_participants: List[str]
    open: bool
    match_ids: List[List[str]]
    name: str
    status: str
    rounds_to_win: int

class CreateTournamentRequest(BaseModel):
    name: str
    participants: List[str]
    open: bool
    rounds_to_win: int

class JoinTournamentRequest(BaseModel):
    owner: str
    name: str

File: client/src/models/Tournament.ts
class TournamentStats {
    username: string;
    wins: number;
    losses: number;
    matches: number;
    points: number;
    constructor(username: string, wins: number, losses: number, matches: number, points: number){
        this.username = username;
        this.wins = wins;
        this.losses = losses;
        this.matches = matches;
        this.points = points;
    }
}

export class Tournament {
    owner: string;
    participants: string[];
    confirmed_participants: string[];
    open: boolean;
    name: string;
    type: string;
    status: string;
    match_ids: string[];
    rounds_to_win: number;
    stats: TournamentStats[];

    constructor(
        owner: string,
        participants: string[],
        open: boolean,
        name: string,
        type: string,
        status: string,
        match_ids: string[],
        rounds_to_win: number
    ){
        this.owner = owner;
        this.participants = participants;
        this.confirmed_participants = [];
        this.open = open;
        this.name = name;
        this.type = type;
        this.status = status;
        this.match_ids = match_ids;
        this.rounds_to_win = rounds_to_win;
        this.stats = [];
    }
};